CharlieBackend.Api.UnitTes/LessonServiceTests.cs

```diff
@@ -57,74 +124,111 @@ public async Task CreateLesson_ReturnLessonDTo_NotEmpty()
-            studentGroupRepositoryMock.Setup(x => x.GetByIdAsync(studentGroup.Id)).ReturnsAsync(studentGroup);
-            studentGroupRepositoryMock.Setup(x => x.GetGroupStudentsIds(studentGroup.Id)).ReturnsAsync(new List<long?> { 11, 14 });
+            var lessonService = new LessonService(
+                unitOfWork: _unitOfWorkMock.Object, 
+                mapper: _mapper,
+                currentUserService: _currentUserServiceMock.Object);
 
-            var visitRepositoryMock = new Mock<IVisitRepository>();
+            //Act
+            var result = await lessonService.CreateLessonAsync(createLessonDTO);
 
+            //Assert
+            result.Should().NotBeNull();
+        }
+        
+        [Fact]
+        public async Task CreateLessonAsync_NonExistingMentorId_ShouldReturnNotFound()
+        {
+            //Arrange
+            #region DATA
+            List<VisitDto> visitDto;
+            LessonDto createdLesson;
+            CreateLessonDto createLessonDto;
+            StudentGroup studentGroup = CreateLessonStudentGroup(out visitDto, out createdLesson, out createLessonDto);
 
-            _unitOfWorkMock.Setup(x => x.LessonRepository).Returns(lessonRepositoryMock.Object);
-            _unitOfWorkMock.Setup(x => x.ThemeRepository).Returns(themeRepositoryMock.Object);
-            _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(mentorReposutoryMock.Object);
-            _unitOfWorkMock.Setup(x => x.StudentGroupRepository).Returns(studentGroupRepositoryMock.Object);
-            _unitOfWorkMock.Setup(x => x.VisitRepository).Returns(visitRepositoryMock.Object);
+            Mentor mentor = new Mentor() { Id = 2 };
+            Mentor mentorWrong = new Mentor() { Id = 31 };
+            var createLessonDtoWrongMentor = new CreateLessonDto
+            {
+                ThemeName = "ExampleName",
+                MentorId = mentorWrong.Id,
+                StudentGroupId = 3,
+                LessonVisits = visitDto
+            };
+            #endregion
 
-            _currentUserServiceMock = GetCurrentUserAsExistingStudent();
+            #region MOCK
+            var mentorReposutoryMockWrong = new Mock<IMentorRepository>();
+            _mentorRepositoryMock.Setup(x => x.GetMentorByIdAsync(mentor.Id)).ReturnsAsync(mentor);
+            mentorReposutoryMockWrong.Setup(x => x.GetMentorByIdAsync(mentorWrong.Id)).ReturnsAsync(default(Mentor));
+
+            MockEntities(studentGroup);
 
             var lessonService = new LessonService(
-                unitOfWork: _unitOfWorkMock.Object, 
+               unitOfWork: _unitOfWorkMock.Object,
+               mapper: _mapper,
+               currentUserService: _currentUserServiceMock.Object);
+
+            #endregion
+            //Act 
+            var result = await lessonService.CreateLessonAsync(createLessonDto);
+            var resultWihtWrongMentor = await lessonService.CreateLessonAsync(createLessonDtoWrongMentor);
```

Comment:
As we test method and one specific state then we have to have only one ACT in each test.
Please write new test for testing expected MentorId.

Reply:
done